robot_calibration/src/capture_features.cpp

```diff
@@ -0,0 +1,304 @@
+// Author: Jack Center
+
+#include <fstream>
+#include <ctime>
+
+#include <ros/ros.h>
+#include <rosbag/bag.h>
+#include <rosbag/view.h>
+#include <rosbag/query.h>
+
+#include <std_msgs/String.h>
+#include <robot_calibration_msgs/CalibrationData.h>
+#include <robot_calibration_msgs/CaptureConfig.h>
+
+#include <robot_calibration/capture/chain_manager.h>
+#include <robot_calibration/capture/feature_finder_loader.h>
+
+#include <camera_calibration_parsers/parse.h>
+#include <robot_calibration/ceres/optimizer.h>
+#include <robot_calibration/camera_info.h>
+#include <robot_calibration/load_bag.h>
+
+#include <boost/foreach.hpp>  // for rosbag iterator
+
+#include <robot_calibration/capture_features.h>
+
+int main(int argc, char** argv)
+{
+  ros::init(argc, argv,"robot_calibration");
+  ros::NodeHandle nh("~");
+
+  if (!check_parameters(nh))
```

Comment:
I assume this is the one from capture_features.h.  You should use namespaces to differentiate functions if there are multiples with the same name.  If you put everything in capture_features in a namespace you could do something like this at the top of this file and it would still work the same (it would just be much more obvious which functions are being called):

```c++
using capture_features::check_parameters;
```